include/vctr/Expressions/Trigonometric/Acos.h

```diff
@@ -0,0 +1,90 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2023 by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::expressions
+{
+
+template <size_t extent, class SrcType>
+requires is::realOrComplexFloatNumber<ValueType<SrcType>>
+class Acos : ExpressionTemplateBase
+{
+public:
+    using value_type = ValueType<SrcType>;
+
+    VCTR_COMMON_UNARY_EXPRESSION_MEMBERS (Acos, src)
+
+    VCTR_FORCEDINLINE constexpr value_type operator[] (size_t i) const
+    {
+#if VCTR_USE_GCEM
+        if constexpr (is::realFloatNumber<value_type>)
+        {
+            if (std::is_constant_evaluated())
```

Comment:
why is it necessary to request this? i mean even if this is not constant evaluated, we could use gcem, right?

Reply:
gcem proved to be slower compared to the std implementation in non constant evaluated cases in the past. Furthermore, in contrast to the std version, it unfortunately does not support complex numbers, so we need the std fallback version in any case. This seemed to be the most straightforward way of chosing the appropriate implementation to me